### PR TITLE
make categories searchable

### PR DESCRIPTION
**Pull request by haesleinhuepf**
Hey Stephane @StRigaud ,

this change is required to make the napari-assistant find the right functions. Under the hood, I read out all functions, that claim to be assistant-compatible like this `list(cle.operations("in assistant").keys())`

I had to modify this:

```
@plugin_function(category=[...
```

to this in many places:
```
@plugin_function(categories=[...
```

Does this require a change in a code-generator? If so, where?

Let me know what you think! (Or feel free to merge this).

Best,
Robert

### CODE REVIEW

**Comment by StRigaud** [comment]
Hey @haesleinhuepf 

Yes, it does, but very minimalist has we just change the name when we write the file:
https://github.com/clEsperanto/gencle/blob/221c044ac6bec00726eafee80ef2f6e7852031ce/update_scripts/gencle/_genpy.py#L207

**Comment by StRigaud** [comment]
Also I did an `operations` function in `functionalities.py` but most likely not good compared to yours:
https://github.com/clEsperanto/pyclesperanto/blob/5382c42d8ba926ab4d18828532053de1f6c9ce12/pyclesperanto/_functionalities.py#L266-L269

dont hesitate to remove it

**Comment by haesleinhuepf** [comment]
Just a note: the pre-commit is super annoying

File: pyclesperanto/_categories.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
+def categories():
+
+    from inspect import getmembers, isfunction
+
+    import pyclesperanto as cle
```
Comment (StRigaud) [inline pyclesperanto/_categories.py:5]: Is this the only way? not a big fan of having an `import` of the package inside the package ...

File: pyclesperanto/_categories.py [hunk 2]
```diff
@@ -0,0 +1,17 @@
+def categories():
+
+    from inspect import getmembers, isfunction
```
Comment (StRigaud) [inline pyclesperanto/_categories.py:3]:
```suggestion
from inspect import getmembers, isfunction
 
def categories():   
```

I would put the import outside of the function if possible

**Comment by StRigaud** [comment]
The `categories` function is missing a doc :wink:

**Pull request merged**